src/fetchers/top-languages-fetcher.js

```diff
@@ -84,20 +86,34 @@ async function fetchTopLanguages(username, exclude_repo = []) {
 
       // if we already have the language in the accumulator
       // & the current language name is same as previous name
-      // add the size to the language size.
+      // add the size to the language size and increase repoCount.
       if (acc[prev.node.name] && prev.node.name === acc[prev.node.name].name) {
         langSize = prev.size + acc[prev.node.name].size;
+        repoCount = repoCount + 1;
+      }
+      else {
+        // reset repoCount to 1
+        // language must exist in atleast one repo to be detected
+        repoCount = 1;
       }
       return {
         ...acc,
         [prev.node.name]: {
           name: prev.node.name,
           color: prev.node.color,
           size: langSize,
+          count: repoCount,
         },
       };
     }, {});
 
+
+  Object.keys(repoNodes)
+    .forEach((name) => {
+      // comparison index calculation
+      repoNodes[name].size = Math.pow(repoNodes[name].size, p) * Math.pow(repoNodes[name].count, q);
```

Comment:
I don't think this is correct as setting q to `0` will now break the algorithm and not fall back to the default behaviour (i.e. `size^1*count^0=0`).

Reply:
`a^0 = 1`, not `0`. (For `a != 0`)

Reply:
See line 106 in the [tests](https://github.com/anuraghazra/github-readme-stats/commit/a30cebea9ebb5a20d83dd4afddfaca97d35434bc).

Reply:
Ah of course! I'm clearly not awake enough to review this PR 💤 . But that is out of the way the PR now looks good to me.